engine-tests/src/test/java/org/terasology/rendering/nui/internal/LineTest.java

```diff
@@ -0,0 +1,102 @@
+    }
+
+    @Test
+    public void testRelativeToAbsolute() {
+        for (int i = 0; i < RANDOM_TEST_COUNT; i++) {
+            Rect2i relativeRegion = Rect2i.createFromMinAndMax(shortRnd(), shortRnd(), shortRnd(), shortRnd());
+
+            assertEquals(Line.relativeToAbsolute(relativeRegion, cropRegion),
+                Rect2i.createFromMinAndSize(cropRegion.minX() + relativeRegion.minX(), cropRegion.minY() + relativeRegion.minY(),
+                    relativeRegion.width(), relativeRegion.height()));
+        }
+    }
+
+    @Test
+    public void testLineCoordinatesNoIntersection() {
+        for (int i = 0; i < RANDOM_TEST_COUNT; i++) {
+            // Line is located in the bottom left, outside the region
+            int sx = cropRegion.minX() - shortRnd();
+            int sy = cropRegion.minY() - shortRnd();
+            int ex = cropRegion.minX() - shortRnd();
+            int ey = cropRegion.minY() - shortRnd();
+
+            assertNull(Line.getLineCoordinates(sx, sy, ex, ey, Rect2i.EMPTY, cropRegion));
+
+            // Top left
+            sx = cropRegion.minX() - shortRnd();
+            sy = cropRegion.maxY() + shortRnd();
+            ex = cropRegion.minX() - shortRnd();
+            ey = cropRegion.maxY() + shortRnd();
+
+            assertNull(Line.getLineCoordinates(sx, sy, ex, ey, Rect2i.EMPTY, cropRegion));
+
+            // Bottom right
+            sx = cropRegion.maxX() + shortRnd();
+            sy = cropRegion.minY() - shortRnd();
+            ex = cropRegion.maxX() + shortRnd();
+            ey = cropRegion.maxY() - shortRnd();
+
+            assertNull(Line.getLineCoordinates(sx, sy, ex, ey, Rect2i.EMPTY, cropRegion));
+
+            // Top right
+            sx = cropRegion.maxX() + shortRnd();
+            sy = cropRegion.maxY() + shortRnd();
+            ex = cropRegion.maxX() + shortRnd();
+            ey = cropRegion.maxY() + shortRnd();
+
+            assertNull(Line.getLineCoordinates(sx, sy, ex, ey, Rect2i.EMPTY, cropRegion));
+        }
+    }
+
+    @Test
+    public void testLineCoordinatesIntersection() {
+        // Test several preset intersecting lines
+        assertEquals(new Line.LineCoordinates(new Vector2i(10, 30), new Vector2i(30, 20)),
+            Line.getLineCoordinates(0, 40, 40, 0, Rect2i.EMPTY, cropRegion));
+        assertEquals(new Line.LineCoordinates(new Vector2i(10, 25), new Vector2i(30, 25)),
+            Line.getLineCoordinates(5, 25, 35, 25, Rect2i.EMPTY, cropRegion));
+        assertEquals(new Line.LineCoordinates(new Vector2i(20, 20), new Vector2i(20, 30)),
+            Line.getLineCoordinates(20, 5, 20, 35, Rect2i.EMPTY, cropRegion));
+        assertEquals(new Line.LineCoordinates(new Vector2i(20, 25), new Vector2i(30, 30)),
+            Line.getLineCoordinates(20, 25, 40, 40, Rect2i.EMPTY, cropRegion));
+    }
+
+    /**
+     * @return A random integer ranging from 1 to 5 (inclusive)
+     */
+    private int shortRnd() {
+        return (int) (Math.random() * 5) + 1;
```

Comment:
I would strongly recommend not to use random in unit tests. You will never know that way if it truly will always pass or if you went just lucky in the test runs you made.